infra/auth-store/src/main/java/com/simprints/infra/authstore/domain/LoginInfoStore.kt

```diff
@@ -26,86 +30,105 @@ internal class LoginInfoStore @Inject constructor(
         private const val CORE_FIREBASE_API_KEY = "CORE_FIREBASE_API_KEY"
     }
 
-    // TODO Move data to an encrypted version - CORE-2590
-    private val prefs: SharedPreferences = ctx.getSharedPreferences(PREF_FILE_NAME, PREF_MODE)
+    @Deprecated("Data has been migrated to secure prefs. Delete after there are no users below 2024.1.0")
+    private val prefs: SharedPreferences =
+        ctx.getSharedPreferences(LEGACY_PREF_FILE_NAME, Context.MODE_PRIVATE)
+
+    private val securePrefs = securityManager.buildEncryptedSharedPreferences(SECURE_PREF_FILE_NAME)
+
+    /**
+     * Ensures that data has been migrated to secure prefs before accessing it.
+     */
+    private fun getSecurePrefs(): SharedPreferences {
+        if (prefs.contains(PROJECT_ID)) {
```

Comment:
Wonder if this would have a non-negligible impact on performance - each time a value is read or saved, the OS will open two files instead of one. Maybe we can reverse this and check if `PROJECT_ID` is present in secure prefs and only proceed to migration (and hence opening the legacy file) if it's missing.

Reply:
I was wondering the same and implemented "in-memory" cache for project and user ID fields, assuming that those probably are the hottest paths. There was no perceivable difference in startup times and app work in general. 

I think the main overhead comes from creating/opening the encrypted file, then the reference is stored in memory (since this is a singleton) and the read/writes are pretty quick.